moveit_ros/moveit_servo/src/collision_check.cpp

```diff
@@ -91,33 +97,35 @@ planning_scene_monitor::LockedPlanningSceneRO CollisionCheck::getLockedPlanningS
 
 void CollisionCheck::start()
 {
-  timer_ = nh_.createTimer(period_, &CollisionCheck::run, this);
+  timer_ = node_->create_wall_timer(std::chrono::duration<double>(period_), std::bind(&CollisionCheck::run, this));
 }
 
 void CollisionCheck::stop()
 {
-  timer_.stop();
+  timer_->cancel();
 }
 
-void CollisionCheck::run(const ros::TimerEvent& timer_event)
+void CollisionCheck::run()
 {
-  // Log warning when the last loop duration was longer than the period
-  if (timer_event.profile.last_duration.toSec() > period_.toSec())
-  {
-    ROS_WARN_STREAM_THROTTLE_NAMED(ROS_LOG_THROTTLE_PERIOD, LOGNAME,
-                                   "last_duration: " << timer_event.profile.last_duration.toSec() << " ("
-                                                     << period_.toSec() << ")");
-  }
+  // // Log warning when the last loop duration was longer than the period
+  // if (timer_event.profile.last_duration.toSec() > period_.toSec())
+  // {
+  //   ROS_WARN_STREAM_THROTTLE_NAMED(ROS_LOG_THROTTLE_PERIOD, LOGNAME,
+  //                                  "last_duration: " << timer_event.profile.last_duration.toSec() << " ("
+  //                                                    << period_.toSec() << ")");
+  // }
 
   if (paused_)
   {
     return;
   }
 
   // Copy the latest joint state
-  auto latest_joint_state = joint_state_subscriber_->getLatest();
-  for (std::size_t i = 0; i < latest_joint_state->position.size(); ++i)
-    current_state_->setJointPositions(latest_joint_state->name[i], &latest_joint_state->position[i]);
+  {
+    const std::lock_guard<std::mutex> lock(joint_state_mutex_);
```

Comment:
why the mutex when the previous version didn't need it?

Reply:
The previous had another class `joint_state_subscriber_` that had the mutex in it. I removed this as Tyler and I added it while debugging kinetic problems in ROS1